File: outdated_codes/Calculator.cpp
#include <iostream>
using namespace std;

float add(float x, float y) {
    return x + y;
}

float subtract(float x, float y) {
    return x - y;
}

float multiply(float x, float y) {
    return x * y;
}

float divide(float x, float y) {
    if (y == 0) {
        return -1;
    } else {
        return x / y;
    }
}

int main() {
    int choice;
    float num1, num2;

    cout << "Simple C++ Calculator" << endl;
    cout << "Select operation:" << endl;
    cout << "1. Add" << endl;
    cout << "2. Subtract" << endl;
    cout << "3. Multiply" << endl;
    cout << "4. Divide" << endl;

    cout << "Enter choice (1/2/3/4): ";
    cin >> choice;
    cout << "Enter first number: ";
    cin >> num1;
    cout << "Enter second number: ";
    cin >> num2;

    if (choice == 1) {
        cout << num1 << " + " << num2 << " = " << add(num1, num2) << endl;
    } else if (choice == 2) {
        cout << num1 << " - " << num2 << " = " << subtract(num1, num2) << endl;
    } else if (choice == 3) {
        cout << num1 << " * " << num2 << " = " << multiply(num1, num2) << endl;
    } else if (choice == 4) {
        if (num2 == 0) {
            cout << "Error! Division by zero." << endl;
        } else {
            cout << num1 << " / " << num2 << " = " << divide(num1, num2) << endl;
        }
    } else {
        cout << "Invalid input. Please select a valid operation." << endl;
    }

    return 0;
}

File: outdated_codes/Calculator.pl
use strict;
use warnings;

sub add {
    return $_[0] + $_[1];
}

sub subtract {
    return $_[0] - $_[1];
}

sub multiply {
    return $_[0] * $_[1];
}

sub divide {
    if ($_[1] == 0) {
        return "Error! Division by zero.";
    } else {
        return $_[0] / $_[1];
    }
}

print "Simple Perl Calculator\n";
print "Select operation:\n";
print "1. Add\n";
print "2. Subtract\n";
print "3. Multiply\n";
print "4. Divide\n";

print "Enter choice (1/2/3/4): ";
my $choice = <STDIN>;
chomp($choice);
print "Enter first number: ";
my $num1 = <STDIN>;
chomp($num1);
print "Enter second number: ";
my $num2 = <STDIN>;
chomp($num2);

if ($choice == 1) {
    print "$num1 + $num2 = ", add($num1, $num2), "\n";
} elsif ($choice == 2) {
    print "$num1 - $num2 = ", subtract($num1, $num2), "\n";
} elsif ($choice == 3) {
    print "$num1 * $num2 = ", multiply($num1, $num2), "\n";
} elsif ($choice == 4) {
    print "$num1 / $num2 = ", divide($num1, $num2), "\n";
} else {
    print "Invalid input. Please select a valid operation.\n";
}
